v3/src/data-interactive/handlers/di-handler-utils.ts

```diff
@@ -55,4 +61,13 @@ export function updateAttribute(attribute: IAttribute, value: DIAttribute, dataC
       metadata?.setIsHidden(attribute.id, value.hidden)
     }
   }
+  if (value?.colormap != null) {
+    if (dataContext) {
+      const metadata = getSharedCaseMetadataFromDataset(dataContext)
+      const categorySet = metadata?.getCategorySet(attribute.id)
+      Object.entries(value?.colormap as DICategoryColorMap || {}).forEach(([category, color]) => {
+        categorySet?.setColorForCategory(category, color || categorySet.colorMap[category] || "")
```

Comment:
```suggestion
        categorySet?.setColorForCategory(category, color)
```
Assuming you can remove `undefined` as a possible type from `DICategoryColorMap` values.